src/main/java/muller/task/Task.java

```diff
@@ -0,0 +1,149 @@
+     * Sets the type of the task (Todo, Deadline, or Event).
+     *
+     * @param type The type of the task, represented as a single character (T, D, E).
+     */
+    public void setType(String type) {
+        this.type = "[" + type + "]";
+    }
+
+    /**
+     * Sets the date for Deadline tasks.
+     *
+     * @param date The date to set.
+     */
+    public void setDate(LocalDate date) {
+        this.date = date;
+    }
+
+    /**
+     * Sets the start and end dates for Event tasks.
+     *
+     * @param startDate The start date of the event.
+     * @param endDate   The end date of the event.
+     */
+    public void setDateRange(LocalDate startDate, LocalDate endDate) {
+        this.date = startDate;
+        this.endDate = endDate;
+    }
+
+    /**
+     * Marks the task as done.
+     */
+    public void markAsDone() {
+        this.isDone = true;
+    }
+
+    /**
+     * Marks the task as done or not done.
+     *
+     * @param isDone Whether the task is done.
+     */
+    public void markAsDone(boolean isDone) {
+        this.isDone = isDone;
+    }
+
+    /**
+     * Marks the task as not done.
+     */
+    public void markAsNotDone() {
+        this.isDone = false;
+    }
+
+    /**
+     * Checks if the task is on a specified date.
+     *
+     * @param date The date to check.
+     * @return True if the task occurs on the specified date, false otherwise.
+     */
+    public boolean isOnDate(LocalDate date) {
+        if (type.equals("[T]")) {
+            return false; // Todo tasks don't have dates
+        } else if (type.equals("[D]")) {
+            return this.date != null && this.date.equals(date);
+        } else if (type.equals("[E]")) {
+            return this.date != null
+                    && this.endDate != null && (
+                        date.equals(this.date)
+                            || date.equals(this.endDate) || (
+                                date.isAfter(this.date) && date.isBefore(this.endDate)));
+        }
+        return false;
+    }
+
+    /**
+     * Returns the icon representing whether the task is done.
+     *
+     * @return The icon representing the task's completion status.
+     */
+    public String doneIcon() {
+        return (isDone ? "[X]" : "[ ]");
+    }
+
+    @Override
+    public String toString() {
+        String dateStr = "";
+        if (type.equals("[D]") && date != null) {
+            dateStr = " (by: " + date.format(OUTPUT_DATE_FORMATTER) + ")";
+        } else if (type.equals("[E]") && date != null && endDate != null) {
+            dateStr = " (from: " + date.format(OUTPUT_DATE_FORMATTER)
+                    + " to: " + endDate.format(OUTPUT_DATE_FORMATTER) + ")";
+        }
+        return this.type + doneIcon() + " " + name + dateStr;
+    }
+
+    /**
+     * Converts the task to a string format suitable for saving to a file.
+     *
+     * @return The string representation of the task for saving.
+     */
+    public String toFileString() {
```

Comment:
This could be changed to a more descriptive method name such as convertToFileString.